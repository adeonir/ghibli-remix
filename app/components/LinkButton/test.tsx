import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import { LinkButton } from '.'

describe('<LinkButton />', () => {
  it(`should render correctly`, () => {
    const { container } = render(
      <LinkButton to="pathname">Children text</LinkButton>,
      {
        wrapper: MemoryRouter,
      }
    )

    expect(
      screen.getByRole('link', { name: /Children text/i })
    ).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should pass the "to" prop correctly', () => {
    render(<LinkButton to="/pathname">Children text</LinkButton>, {
      wrapper: MemoryRouter,
    })

    expect(
      screen.getByRole('link', { name: /Children text/i })
    ).toHaveAttribute('href', '/pathname')
  })
})
