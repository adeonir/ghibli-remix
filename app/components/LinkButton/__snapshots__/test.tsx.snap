// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
<div>
  <a
    class="inline-block rounded-lg bg-teal-500 px-4 py-2 font-bold text-slate-100 transition hover:bg-teal-550 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-teal-500 disabled:cursor-not-allowed disabled:opacity-75"
    href="/pathname"
  >
    Children text
  </a>
</div>
`;
